packages/workflow/src/agent/agent-workflow.ts

```
@@ -57,6 +56,24 @@ export class AgentStepEvent extends WorkflowEvent<{
   toolCalls: AgentToolCall[];
 }> {}
 
+export type AgentWorkflowParams = {
+  /**
+   * List of agents to include in the workflow.
+   * Need at least one agent.
+   */
+  agents: BaseWorkflowAgent[];
```

you use a agent's name for handoff - what if you pass two agents with the same name?

---

We store the agents as a map, but allow passing them as a list for convenience. As a result, the new one will replace the existing one with the same name.

---

the constructor should throw an error if there are two agents with the same name as this would lead to unexpected behavior